### ISSUE TITLE
Discuss policy on camelCase in QtSubclasses, then add to docs

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#1091]
## 📚 Documentation
@jni asked in  https://github.com/napari/napari/pull/1023#discussion_r399197252
>Do we have have a policy on how we deal with camelCase vs under_scores when adding functions to Qt subclasses? Whatever it is, we should put it in the contributing guide.

I don't think there is a formal policy... but we can discuss what it might be here and then add it to contributing.md

**Comment by sofroniewn** [comment]
See also #6 our oldest open issue from @kne42!!! I'd love to resolve this, standardize the repo and update the contributing guidelines. I don't come into this with strong opinions, so if anyone wants to put out a proposal that would be great!!!

**Comment by tlambert03** [comment]
interesting!  I also don't really have strong opinions either way, though my habit had been to follow #6.  It was my impression though that @jni had the opposite preference to what @kne42  was saying there in #6 (i.e. our stuff should always be pythonic) ... but I could have misinterpreted that.

**Comment by tlambert03** [comment]
Would be nice to get something like a show of hands on this @napari/core-devs:

1. Stick with camelCase for method names in Qt Subclasses. (see [`QRangeSlider`](https://github.com/napari/napari/blob/master/napari/_qt/qt_range_slider.py#L13) for an example)
2. Always prefer snake_case for our own method names (see [`QtLayerList`](https://github.com/napari/napari/blob/master/napari/_qt/qt_layerlist.py#L19) for an example)
   - downside: because some of these subclasses will be reimplementing Qt methods, this will inevitably lead to the case where we have a mix of camelCase (for reimplemented methods) and snake_case (for our added methods).  But we could probably create a test that says "if the superclass doesn't have this method, make sure it's snake case)
   - upside: it's more pythonic... and in #1376 we are about to rely on the assumption that event handler methods are named `_on_something_change`... so we're more or less demanding at least partial snake_case names.

If people want to just reply "`#1`" or "`#2`", I can maybe start on writing a test and standardizing

**Comment by sofroniewn** [comment]
Yeah, the #1376 case is a little weird, otherwise I would have wanted to go with option 1.

@ziyangczi recently suggested we go with a function annotation approach instead of a naming convention in #1376, which could be quite nice and mean that we can have camelCase is all Qt Subclasses. If anyone has a concrete proposal for the function annotation case for #1376 I could do it over there now before merge.

How about people respond celebration for 1, rocket ship for 2, and heart for 1 if we fix 1376? Can have an emoji poll :-)

**Comment by tlambert03** [comment]
how do express my ❤️ for that comment without voting?

**Comment by sofroniewn** [comment]
> how do express my ❤️ for that comment without voting?

It's rigged, like one of those twitter polls which is `like` vs `retweet` 🤣
